substrate/primitives/runtime/src/proving_trie.rs

```diff
@@ -0,0 +1,339 @@
+// This file is part of Substrate.
```

Comment:
@Ank4n if you'd like to explore trie-based migrations for AH, this file is of interest ;)